control_panel_api/tests/test_services.py

```diff
@@ -0,0 +1,84 @@
+import json
+from unittest.mock import patch, call
+
+from django.test.testcases import SimpleTestCase, override_settings
+
+from control_panel_api import services
+
+
+@override_settings(ENV='test', BUCKET_REGION='eu-test-2', LOGS_BUCKET_NAME='moj-test-logs')
+class ServicesTestCase(SimpleTestCase):
+    def test_policy_document_readwrite(self):
+        document = services.get_policy_document('foo', readwrite=True)
+
+        sids = [s['Sid'] for s in document['Statement']]
+        self.assertIn('UpdateRenameAndDeleteObjects', sids)
+
+        document = services.get_policy_document('foo', readwrite=False)
+
+        sids = [s['Sid'] for s in document['Statement']]
+        self.assertNotIn('UpdateRenameAndDeleteObjects', sids)
+
+    @patch('boto3.client')
+    def test_create_bucket(self, mock_client):
+        services.create_bucket('bucketname')
+
+        mock_client.return_value.create_bucket.assert_called_with(
+            Bucket='test-bucketname',
+            ACL='private',
+            CreateBucketConfiguration={
+                'LocationConstraint': 'eu-test-2',
+            },
+        )
+        mock_client.return_value.put_bucket_logging.assert_called_with(
+            Bucket='test-bucketname',
+            BucketLoggingStatus={
+                'LoggingEnabled': {
+                    'TargetBucket': 'moj-test-logs',
+                    'TargetPrefix': 'test-bucketname/'
+                }
+            }
+        )
+
+    @patch('boto3.client')
+    def test_delete_bucket(self, mock_client):
+        services.delete_bucket('bucketname')
+
+        mock_client.return_value.delete_bucket.assert_called_with(
+            Bucket='test-bucketname',
+        )
+
+    @patch('boto3.client')
+    def test_create_bucket_policies(self, mock_client):
+        services.create_bucket_policies('bucketname')
+
+        expected_calls = [
+            call(
+                PolicyName='test-bucketname-readwrite',
+                PolicyDocument=json.dumps(services.get_policy_document('test-bucketname', True))
```

Comment:
See other comment: https://github.com/ministryofjustice/analytics-platform-control-panel/pull/16/files#r138314735

Use `services.get_policy_document` (the actual code) means it will always pass even if, let's say we break it big time, we add a `"s3:DeleteObject",` to the `ReadObjects` actions.

We should keep a copy of the expected policies in the tests (even if it's a duplication it's a good one in this case).